File: java/example-supplier/src/main/java/dev/kaiquebt/anycall/example/ProductSupplier.java
package dev.kaiquebt.anycall.example;

import dev.kaiquebt.anycall.annotation.Supply;
import dev.kaiquebt.anycall.example.model.CreateProductRequest;
import dev.kaiquebt.anycall.example.model.Product;

public class ProductSupplier {

    @Supply("create-new-product")
    public Product createNewProduct(CreateProductRequest req) {
        return new Product(req.name(), req.priceInCents());
    }
}

File: python/example-supplier/src/supplier/product_supplier.py
from anycall import supply

from .model.create_product_request import CreateProductRequest
from .model.product import Product


class ProductSupplier:
    """Supplier for product creation operations."""

    @supply("create-new-product")
    def create_new_product(self, req: CreateProductRequest) -> Product:
        """Create a new product.

        Args:
            req: Product creation request

        Returns:
            Created product
        """
        return Product(name=req.name, price_in_cents=req.price_in_cents)
